pkg/node/manager/manager.go

```diff
@@ -651,15 +652,9 @@ func (m *Manager) NodeDeleted(n nodeTypes.Node) {
 			prefix = ip.IPToNetPrefix(address.IP.To16())
 		}
 		m.ipcache.Delete(prefix.String(), n.Source)
-	}
 
-	for _, address := range []net.IP{
-		entry.node.IPv4HealthIP, entry.node.IPv6HealthIP,
-		entry.node.IPv4IngressIP, entry.node.IPv6IngressIP,
-	} {
-		if address != nil {
-			m.ipcache.Delete(address.String(), n.Source)
-		}
+		resource := ipcacheTypes.NewResourceID(ipcacheTypes.ResourceKindNode, "", n.Name)
+		m.ipcache.RemoveLabels(prefix, labels.Labels{}, resource)
```

Comment:
Strictly only this change will cause other bugs because we should be calling `RemoveLabels` on the inverse of wherever `upsertIntoIDMD` is called. In other words, `upsertIntoIDMD` and `RemoveLabels` need to be symmetric.

We already have a convenient function analogous to `upsertIntoIDMD` called `deleteIPCache`. I think the call to `RemoveLabels` should go inside `deleteIPCache` without removing the existing ipcache calls.